lib/WebService/PayPal/PaymentsAdvanced/Role/HasTransactionTime.pm

```diff
@@ -10,20 +10,22 @@ use feature qw( state );
 
 use DateTime::TimeZone;
 use DateTime::Format::MySQL;
-use Types::Standard qw( InstanceOf );
+use Types::Standard qw( InstanceOf Maybe );
 
 has transaction_time => (
     is  => 'lazy',
-    isa => InstanceOf ['DateTime'],
+    isa => Maybe [ InstanceOf ['DateTime'] ],
 );
 
 sub _build_transaction_time {
     my $self = shift;
 
     state $time_zone
         = DateTime::TimeZone->new( name => 'America/Los_Angeles' );
-    my $dt = DateTime::Format::MySQL->parse_datetime(
-        $self->params->{TRANSTIME} );
+
+    return undef unless my $transtime = $self->params->{TRANSTIME};
+
+    my $dt = DateTime::Format::MySQL->parse_datetime($transtime);
```

Comment:
MySQL? 😕

Reply:
Yeah, I am not sure about that either, but I guess it works.